toolkit/tools/internal/safemount.go/safemount.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Package that assists with mounting and unmounting cleanly.
+package safemount
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+	"golang.org/x/sys/unix"
+)
+
+type Mount struct {
+	target     string
+	isMounted  bool
+	dirCreated bool
+}
+
+// Creates a new system mount.
+func NewMount(source, target, fstype string, flags uintptr, data string, makeAndDeleteDir bool) (*Mount, error) {
```

Comment:
Thoughts on adding a bit more smarts here to coordinate nested mounts/unmounts (ie `/boot`, `/boot/efi`)? Something I've had bubbling in the back of my brain is to create a standardized way to manage mount points in the imager tool that knows how to cleanup after itself. Not critical for "simple" mounts, but dm-verity and dm-crypt mounts can get a bit complicated.

Reply:
The general rule of thumb is to cleanup in the reverse of the order of setup. Would this be sufficient for dm-verity and dm-crypt?

It might make sense to create a safemount list type. Though, I don't need such a thing in this change.

Reply:
Reverse order may be sufficient, I'm mostly interested in adding smarts to each 'type' of mount that knows their specific cleanup requirements. This is mostly a commentary on how disorganized the imager tool is though.

No need to do this now though, we can revisit later.